src/main.rs

```diff
@@ -301,6 +306,17 @@ fn scan(root: &Path, pattern: Arc<Regex>, base: &Path, config: Arc<FdOptions>) {
                 },
             }
 
+            // Filter out unwanted extensions.
+            match (&config.extension, entry.path().extension()) {
```

Comment:
`entry.path().extension()` dynamically extracts the extension. I imagine it would be better to match/map on `config.extension` alone first, so as to avoid calling it on searches that don't need it.

Reply:
Yes, I also think this would be better.